pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2252,8 +2252,8 @@ func TestListUsersCycleDetection(t *testing.T) {
 				},
 				visitedUsersetsMap: visitedUsersets,
 			}, channelWithResults)
-			if err != nil {
-				channelWithError <- err
+			if resp.err != nil {
+				channelWithError <- resp.err
 				return
 			}
 			channelDone <- struct{}{}
```

Comment:
Can we add this test as well please? It makes sure that if we experience cycles through independent object_ids and with a non-stratifiable resolution on the `document:1#viewer` and `document:2#blocked` relationship then we still yield the residual relationships for `user:jon`.

```go
		{
			name: "exclusion_with_chained_negation",
			req: &openfgav1.ListUsersRequest{
				Object:   &openfgav1.Object{Type: "document", Id: "2"},
				Relation: "viewer",
				UserFilters: []*openfgav1.UserTypeFilter{
					{
						Type: "user",
					},
				},
			},
			model: `model
			  schema 1.1

			type user

			type document
			  relations
			    define unblocked: [user]
				define blocked: [user, document#viewer] but not unblocked
				define viewer: [user, document#blocked] but not blocked
			`,
			tuples: []*openfgav1.TupleKey{
				tuple.NewTupleKey("document:1", "viewer", "document:2#blocked"),
				tuple.NewTupleKey("document:2", "blocked", "document:1#viewer"),
				tuple.NewTupleKey("document:2", "viewer", "user:jon"),
				tuple.NewTupleKey("document:2", "unblocked", "user:jon"),
			},
			expectedUsers: []string{"user:jon"},
		},
```